### DISCUSSION TITLE
Running on CPU instead of GPU in an AMD system.  How to fix?

### DISCUSSION THREAD

**Discussion by morpheuskibbe** [lshqqytiger/stable-diffusion-webui-directml#124] [Q&A] [answered]
Hello.  I have an AMD system.  i got it to run by using these commands

set COMMANDLINE_ARGS=--skip-torch-cuda-test --no-half

Initially because it refused to start with no cuda. hence the skip cuda option and then because it had a runtime error mentioning "LayerNormKernelImpl" not implemented for 'Half'

It's running now but according to task manager it's running on CPU only... the GPU is not being hit in any way and predictably the performance is terrible. 

what commands should I be using to get expected behavior?

**Comment by morpheuskibbe** [comment, marked as answer]
OH MY GOD the instructions were WRONG

The automatic installation section says
git clone https://github.com/AUTOMATIC1111/stable-diffusion-webui.git

It should say
git clone https://github.com/lshqqytiger/stable-diffusion-webui-directml.git

**Reply by robonxt** [reply]
Actually, if you go to the AMD installation guide, it says to use the ishqqytiger fork: https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Install-and-Run-on-AMD-GPUs

**Reply by morpheuskibbe** [reply]
yes. hence my comment about incorrect instructions.  if you goto the ishqqtiger fork: https://github.com/lshqqytiger/stable-diffusion-webui-directml and scroll down to the instructions it has a copyable git clone command that points at the original, not the fork.  Hence my original issue.

**Reply by Enferlain** [reply]
I cloned the directml one and it still runs on cpu 4shrug

**Reply by Aev69** [reply]
Same here. Both the original and fork just run on CPU, really defeats the purpose of using the directml one

**Comment by Ming2k8-Coder** [comment]
What your parameter in webui-user.bat

Vào 09:09, T.2, 8 Th1, 2024 Aev69 ***@***.***> đã viết:

**Comment by diekf** [comment]
try add --use-directml

**Comment by dylanh724** [comment]
Still an issue. Adding `--use-directml` doesn't work, either; heck it's not even detected as a command. Confirming I'm on this fork.

**Reply by lshqqytiger** [reply]
Make sure that you are on the latest version 1.9.3.

**Reply by tonkgai** [reply]
May I ask. version 1.9.3 for what?     I have the same problem!   I followed this one "git clone https://github.com/lshqqytiger/stable-diffusion-webui-directml && cd stable-diffusion-webui-directml && git submodule init && git submodule update"

**Comment by shazam1234raja** [comment]
I have been having the same issue 
I am using SD.next as you see in the picture 
NO GPU showing anywhere 
1 generation takes with Width: 320 Height 320 takes more then 30mins

Running 
windows 10 pro
GPU - AMD Radeon RX 6600 XT
CPU - AMD Ryzen 5 5600 6-Core Processor
VRAM - 8176 MB GDDR6 2000 MHz
RAM - 16 GB


![image](https://github.com/user-attachments/assets/4de86726-3031-4f3b-bee0-f6bdd2d0f605)


Starting SD.Next
17:26:35-998149 INFO     Logger: file="E:\Stable Diffusion\ZLUDA STABLE diffusion\automatic\sdnext.log" level=DEBUG
                         size=65 mode=create
17:26:36-000125 INFO     Python: version=3.10.6 platform=Windows bin="E:\Stable Diffusion\ZLUDA STABLE
                         diffusion\automatic\venv\Scripts\python.exe" venv="E:\Stable Diffusion\ZLUDA STABLE
                         diffusion\automatic\venv"
17:26:36-431118 INFO     Version: app=sd.next updated=2024-12-09 hash=a2f5a39a branch=master
                         url=https://github.com/vladmandic/automatic/tree/master ui=main
17:26:37-020432 INFO     Platform: arch=AMD64 cpu=AMD64 Family 25 Model 33 Stepping 2, AuthenticAMD system=Windows
                         release=Windows-10-10.0.19045-SP0 python=3.10.6 docker=False
17:26:37-024286 DEBUG    Packages: venv=venv site=['venv', 'venv\\lib\\site-packages']
17:26:37-025287 INFO     Args: ['--use-zluda', '--debug', '--autolaunch']
17:26:37-027287 DEBUG    Setting environment tuning
17:26:37-028288 DEBUG    Torch allocator: "garbage_collection_threshold:0.80,max_split_size_mb:512"
17:26:37-034289 DEBUG    Torch overrides: cuda=False rocm=False ipex=False directml=False openvino=False zluda=True
17:26:37-045292 INFO     Python: version=3.10.6 platform=Windows bin="E:\Stable Diffusion\ZLUDA STABLE
                         diffusion\automatic\venv\Scripts\python.exe" venv="E:\Stable Diffusion\ZLUDA STABLE
                         diffusion\automatic\venv"
17:26:37-050315 INFO     ROCm: AMD toolkit detected
17:26:37-499652 INFO     ROCm: agents=['gfx1032']
17:26:37-501652 INFO     ROCm: version=6.2, using agent gfx1032
17:26:37-537660 **WARNING  Failed to load ZLUDA: Could not find module 'E:\Stable Diffusion\ZLUDA STABLE
                         diffusion\automatic\.zluda\nvrtc64_112_0.dll' (or one of its dependencies). Try using the full
                         path with constructor syntax.**
17:26:37-539702 INFO     **Using CPU-only torch**
17:26:37-726465 INFO     Install: verifying requirements
17:26:37-730508 INFO     Verifying packages
17:26:37-817570 DEBUG    Timestamp repository update time: Mon Dec  9 13:57:14 2024
17:26:37-819571 INFO     Startup: standard
17:26:37-820571 INFO     Verifying submodules
17:26:43-252439 DEBUG    Git submodule: extensions-builtin/sd-extension-chainner / main
17:26:43-520436 DEBUG    Git detached head detected: folder="extensions-builtin/sd-extension-system-info" reattach=main
17:26:43-522436 DEBUG    Git submodule: extensions-builtin/sd-extension-system-info / main
17:26:43-710877 DEBUG    Git submodule: extensions-builtin/sd-webui-agent-scheduler / main
17:26:43-984535 DEBUG    Git detached head detected: folder="extensions-builtin/sdnext-modernui" reattach=main
17:26:43-985536 DEBUG    Git submodule: extensions-builtin/sdnext-modernui / main
17:26:44-176897 DEBUG    Git submodule: extensions-builtin/stable-diffusion-webui-rembg / master
17:26:44-366913 DEBUG    Git submodule: modules/k-diffusion / master
17:26:44-638891 DEBUG    Git detached head detected: folder="wiki" reattach=master
17:26:44-640893 DEBUG    Git submodule: wiki / master
17:26:44-758919 DEBUG    Register paths
17:26:44-813931 DEBUG    Installed packages: 182
17:26:44-814931 DEBUG    Extensions all: ['Lora', 'sd-extension-chainner', 'sd-extension-system-info',
                         'sd-webui-agent-scheduler', 'sdnext-modernui', 'stable-diffusion-webui-rembg']
17:26:44-970965 DEBUG    Extension installer: E:\Stable Diffusion\ZLUDA STABLE
                         diffusion\automatic\extensions-builtin\sd-webui-agent-scheduler\install.py
17:26:46-277078 DEBUG    Extension installer: E:\Stable Diffusion\ZLUDA STABLE
                         diffusion\automatic\extensions-builtin\stable-diffusion-webui-rembg\install.py
17:26:55-136993 DEBUG    ONNX: version=1.20.1 provider=CPUExecutionProvider, available=['AzureExecutionProvider',
                         'CPUExecutionProvider']
17:26:55-179003 INFO     **Device**:

**Reply by lshqqytiger** [reply]
`main` branch does not support HIP SDK 6.2. Try `dev` branch with new venv.

**Reply by shazam1234raja** [reply]
can you please simplify the answer  

![image](https://github.com/user-attachments/assets/a409dcb7-87b8-4b88-87ba-949c1d87de90)

benchmark problem
